src/scheduler.cpp

```diff
@@ -412,6 +412,15 @@ void Scheduler::Worker::stop() {
   }
 }
 
+bool Scheduler::Worker::wait(const TimePoint* timeout) {
+  DBG_LOG("%d: WAIT(%d)", (int)id, (int)currentFiber->id);
+  {
+    std::unique_lock<std::mutex> lock(work.mutex);
+    suspend(timeout);
+  }
+  return timeout == nullptr || std::chrono::system_clock::now() < *timeout;
```

Comment:
Is there no danger of returning 'true' when you've completed successfully, but very close to the timeout time?

Reply:
There is, but when you have timeouts that close, what difference does it make? There should be no side effects to the predicate, so a false timeout should be harmless and indistinguishable from a true timeout.

Reply:
I suppose, but I guess I would have expected suspend(timeout) to return whether or not it timed out, and to use that in the return value. If a task completes before it times out, but then there's a delay and you return that it did time out, it would be odd/confusing. We'd have to worry about padding the timeout for the window between the end of the task completing successfully, and Scheduler::Worker::wait returning.

This being said, I'm not opposed to accepting this as is. Just wanted to hear your thoughts.